heat/utils/data/datatools.py

```diff
@@ -0,0 +1,211 @@
+import h5py
+import numpy as np
+import base64
+import os
+import torch
+from torch.utils import data as torch_data
+from typing import Callable, Iterator, Union
+
+from heat.core import dndarray
+
+__all__ = ["DataLoader", "Dataset"]
+
+
+class DataLoader:
+    """
+    Data Loader. The combines either a ``DNDarray`` or a torch ``Dataset`` with a sampler. It will provide an iterable
+    over the local dataset and it will have a ``shuffle()`` function which calls the ``shuffle()`` function of the
+    given Dataset. If a HeAT DNDarray is given a general Dataset will be created.
+
+    Currently, the DataLoader supports only map-style datasets with single-process loading, and users the random
+    batch sampler. The rest of the DataLoader functionality mentioned in :func:`torch.utils.data.dataloader` applies.
+
+    Arguments:
+        data : Dataset or DNDarray
+            dataset from which to load the data.
+        batch_size : int, optional
+            how many samples per batch to load (default: ``1``).
+        num_workers : int, optional
+            how many subprocesses to use for data loading. ``0`` means that the data will be loaded in the main process.
+            (default: ``0``)
+        collate_fn : callable, optional
+            merges a list of samples to form a mini-batch of Tensor(s).  Used when using batched loading from a
```

Comment:
Torch Tensors or DNDarrays? If DNDarray, please use that or array to avoid confusion